lib/s3.js

```diff
@@ -141,4 +140,30 @@ module.exports = CoreObject.extend({
       });
     });
   },
+
+  listAllObjects: function(options) {
+    var client         = this._client;
+    var listObjects    = Promise.denodeify(client.listObjects.bind(client));
+    var allRevisions   = [];
+
+    return new Promise(function(resolve, reject) {
```

Comment:
Yeah - I prefer that. 

I originally wrote this without the `denodeify`ed `listObjects` hence the shape...